builtin/providers/google/resource_container_cluster.go

```diff
@@ -0,0 +1,383 @@
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"network_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "default",
+				ForceNew: true,
+			},
+
+			"node_config": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"machine_type": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+							ForceNew: true,
+						},
+
+						"service_account": &schema.Schema{
+							Type:     schema.TypeList,
+							Optional: true,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"email": &schema.Schema{
+										Type:     schema.TypeString,
+										Required: true,
+										ForceNew: true,
+									},
+									"scopes": &schema.Schema{
+										Type:     schema.TypeList,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+										Required: true,
+										ForceNew: true,
+									},
+								},
+							},
+						},
+
+						"source_image": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+							ForceNew: true,
+						},
+					},
+				},
+				Set: func(v interface{}) int {
+					// there can be only a single node_config; no need to hash anything
+					return 0
+				},
+			},
+
+			"number_of_nodes": &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"instance_group_urls": &schema.Schema{
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceContainerClusterCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	zoneName := d.Get("zone").(string)
+	clusterName := d.Get("name").(string)
+
+	masterAuths := d.Get("master_auth").(*schema.Set).List()
+	if len(masterAuths) > 1 {
+		return fmt.Errorf("Cannot specify more than one master_auth.")
+	}
+	masterAuth := masterAuths[0].(map[string]interface{})
+
+	cluster := &container.Cluster{
+		MasterAuth: &container.MasterAuth{
+			Password: masterAuth["password"].(string),
+			User:     masterAuth["username"].(string),
+		},
+		Name:     clusterName,
+		NumNodes: int64(d.Get("number_of_nodes").(int)),
+	}
+
+	if v, ok := masterAuth["bearer_token"]; ok {
+		cluster.MasterAuth.BearerToken = v.(string)
+	}
+
+	if v, ok := d.GetOk("cluster_api_version"); ok {
+		cluster.ClusterApiVersion = v.(string)
+	}
+
+	if v, ok := d.GetOk("container_ipv4_cidr"); ok {
+		cluster.ContainerIpv4Cidr = v.(string)
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		cluster.Description = v.(string)
+	}
+
+	if v, ok := d.GetOk("enable_cloud_logging"); ok {
+		cluster.EnableCloudLogging = v.(bool)
+	}
+
+	if v, ok := d.GetOk("network_name"); ok {
```

Comment:
I was merely trying to be explicit (after learning my lessons from some AWS resources which have many different unique IDs), but looking at https://www.terraform.io/docs/providers/google/r/compute_network.html it's not necessary since GCloud doesn't have anything like IDs, so the unique ID will always be name (?). Correct me if I'm wrong.

If the above is true, I'm totally fine with changing this to `network`.